### PR TITLE
Allow declare language prefix on step go to

### LINKED ISSUE #124: Allow to declare the language prefix on the step go to
If I use the step `When I go to the "node" entity with label "Behat recipe to translate ES"` the resulting path is http://domain/node/848, and the system will be redirected to the default language if is English the path will be http://domain/en/node/848.

So I suggest allowing to declare the language on the step as an optional parameter.
Example:
`Given I go to the "node" entity with label "Behat recipe to translate ES" in "Spanish" language`
The result will be http://domain/es/node/848.

### PR DESCRIPTION
**Pull request by Eduardo-Morales-Alberti**
closes #124

### CODE REVIEW

File: src/Behat/Cores/Drupal8.php [hunk 1]
```diff
@@ -469,4 +469,23 @@ public function createFileUrl($file, bool $relative = TRUE) {
...
+   */
+  public function getLanguagePrefix($language) {
+    $language_manager = \Drupal::languageManager();
+    $language_list = $language_manager->getStandardLanguageList();
+    array_walk($language_list, function (&$item1, $key) {
+      $item1 = current($item1);
+    });
+    $language_list = array_flip($language_list);
+    $langcode = $language_list[$language] ?? '';
+    if ($langcode) {
```
Comment (rsanzante) [inline src/Behat/Cores/Drupal8.php:483]: I think it's clear if the "if" checks for an error and then throws an exception. Ending the function throwing an exception looks weird to me.

```
  if ($langcode === '') {
    throw new \InvalidArgumentException(sprintf("Language %s not found", $language));  
  }

  $prefixes = \Drupal::config('language.negotiation')->get('url.prefixes');
  return $prefixes[$langcode] . '/';

```
The idea is an error condition is checked, and if true throw exception, continue if not.

File: src/Behat/Cores/Drupal8.php [hunk 2]
```diff
@@ -469,4 +469,23 @@ public function createFileUrl($file, bool $relative = TRUE) {
     throw new InvalidArgumentException('%s method only accept %s objects in Drupal 8 or higher', __METHOD__, FileInterface::class);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getLanguagePrefix($language) {
+    $language_manager = \Drupal::languageManager();
+    $language_list = $language_manager->getStandardLanguageList();
+    array_walk($language_list, function (&$item1, $key) {
```
Comment (mistermoper) [inline src/Behat/Cores/Drupal8.php:478]: In this way, if there are two words to define a language, only the first would be used.

Such an example, if Spanish can be selected as 'Spanish' or 'Español' (see [here](https://git.drupalcode.org/project/drupal/-/blob/9.4.x/core/lib/Drupal/Core/Language/LanguageManager.php#L254)), only Spanish can be used in the step. If there are multiple ways to write the language label we must allow all so testers are free to write it in an intuitive manner.

I suggest building a key value array ($language_list_map) with the key being the label and the value being the lancode, something like this:

```php
  [
     'Spanish' => 'es',
     'Español => 'es',
    ...
  ]
```
Then you can use that variable in the line 482:

```php
     $langcode = $language_list_map[$language] ?? '';
```

**Review by mistermoper — CHANGES_REQUESTED** [review]
https://github.com/Metadrop/behat-contexts/pull/125/files#r778057004

File: src/Behat/Cores/Drupal8.php [hunk 3]
```diff
@@ -469,4 +469,23 @@ public function createFileUrl($file, bool $relative = TRUE) {
...
+      $item1 = current($item1);
+    });
+    $language_list = array_flip($language_list);
+    $langcode = $language_list[$language] ?? '';
+    if ($langcode) {
+      $prefixes = \Drupal::config('language.negotiation')->get('url.prefixes');
+      return $prefixes[$langcode] . '/';
```
Comment (rsanzante) [inline src/Behat/Cores/Drupal8.php:485]: I would return just the $langcode. The caller, if it needs it, would add the '/'.

File: src/Behat/Cores/Drupal8.php (hunk 2 above)
Comment (rsanzante) [inline src/Behat/Cores/Drupal8.php:478]: The process of finding the value seems a little bit complex (walk over elements, flip and check for the language). I would use arra_filter and chek for results.

Adding the two comments below the function would be:
```
function getLanguagePrefix($language) {
  $language_manager = \Drupal::languageManager();
  $language_list = $language_manager->getStandardLanguageList();

  $filter_func = function($item) use ($language) { 
    return $language === $item[0];
  };
  
  $found = array_filter($language_list, $filter_func);
  
  if (empty($found)) {
    throw new \InvalidArgumentException(sprintf("Language %s not found", $language));  
  }

  $prefixes = \Drupal::config('language.negotiation')->get('url.prefixes');
  return $prefixes[array_key_first($found)];
}

```

**Review by rsanzante — CHANGES_REQUESTED** [review]

File: src/Behat/Cores/Drupal8.php (hunk 2 above)
Comment (mistermoper) [reply src/Behat/Cores/Drupal8.php:478]: I would change this
```
    return $language === $item[0];
```

by this:

```
    return in_array($language, $item);
```

In this comment I explain why: https://github.com/Metadrop/behat-contexts/pull/125#discussion_r778057004

**Comment by Eduardo-Morales-Alberti** [comment]
@rsanzante @mistermoper Requested changes done and tested on local.

**Review by mistermoper — APPROVED** [review]
LGTM

**Review by rsanzante — APPROVED** [review]

**Pull request merged**